src/libra_py/dynamics/qtag/QTAG_calc.py

```diff
@@ -0,0 +1,164 @@
+"""
+..module:: QTAG_calc
+  :platform: Unix, Windows
+  :synopsis: This module contains "ground-level" functions for calculations, mostly output things like energy, norm, etc.
+
+..moduleauthors :: Matthew Dutra
+"""
+
+import sys
+from liblibra_core import *
+
+import numpy as np
+from QTAG_config import univ
+
+def psi(qpas,c,x0):
+	"""Returns the (complex) wavefunction value *wf* at a given point *x0*, calculated using the single-surface basis parameters stored in *qpas* and coefficients *c*.
+
+        Args:
+                qpas (list): List of {q,p,a,s} MATRIX objects.
+
+                c (CMATRIX): The ntraj-by-1 complex matrix of basis coefficients.
+
+		x0 (MATRIX): The matrix of coordinates [x_1,...,x_ndof] at which the wavefunction value should be calculated.
+
+        Returns:
+                wf (complex): Complex value of the wavefunction at x0.
+        """
+
+	qvals,pvals,avals,svals=qpas[0],qpas[1],qpas[2],qpas[3]
+	ndof,ntraj=univ['ndof'],univ['ntraj']
```

Comment:
don't pass the global (imported) parameters - the `univ` should be an input structure